IdokladSdk/Validation/Attributes/NullOrEmptyStringIfAttribute.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.ComponentModel.DataAnnotations;
+using IdokladSdk.Models.Common.Helpers;
+using Microsoft.IdentityModel.Tokens;
+
+namespace IdokladSdk.Validation.Attributes
+{
+    public class NullOrEmptyStringIfAttribute : ValidationAttribute
+    {
+        public NullOrEmptyStringIfAttribute(string dependentProperty, object targetValue)
+        {
+            DependentProperty = dependentProperty;
+            TargetValue = targetValue;
+        }
+
+        private string DependentProperty { get; }
+
+        private object TargetValue { get; }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            var dependentPropertyValue =
+                NullablePropertyHelper.GetDependentPropertyValue(validationContext, DependentProperty);
+
+            if (
+                dependentPropertyValue != null
+                && dependentPropertyValue.Equals(TargetValue)
+                && !((string)value).IsNullOrEmpty())
```

Comment:
V iDokladu máme extension metodu IsNotNullOrEmpty na stringu. Zvážil bych zavedení i tu. Je to přehlednější než vykřičník.